include/pm/device_runtime.h

```diff
@@ -8,107 +9,104 @@
 #define ZEPHYR_INCLUDE_PM_DEVICE_RUNTIME_H_
 
 #include <device.h>
-#include <kernel.h>
-#include <sys/atomic.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
- * @brief Runtime Power Management API
- *
- * @defgroup runtime_power_management_api Runtime Power Management API
+ * @defgroup runtime_power_management_api Device Runtime Power Management API
  * @ingroup power_management_api
  * @{
  */
 
-#ifdef CONFIG_PM_DEVICE_RUNTIME
-
+#if defined(CONFIG_PM_DEVICE_RUNTIME) || defined(__DOXYGEN__)
 /**
  * @brief Enable device runtime PM
  *
- * Called by a device driver to enable device runtime power management.
- * The device might be asynchronously suspended if runtime PM is enabled
- * when the device is not use.
- *
  * @funcprops \pre_kernel_ok
  *
- * @param dev Pointer to device structure of the specific device driver
- * the caller is interested in.
+ * @param dev Device instance.
  */
 void pm_device_enable(const struct device *dev);
 
 /**
  * @brief Disable device runtime PM
  *
- * Called by a device driver to disable device runtime power management.
- * The device might be asynchronously resumed if runtime PM is disabled
+ * If the device is currently suspended it will be resumed.
  *
  * @funcprops \pre_kernel_ok
  *
- * @param dev Pointer to device structure of the specific device driver
- * the caller is interested in.
+ * @param dev Device instance.
+ *
+ * @retval 0 If the device runtime PM is disabled successfully.
+ * @retval -ENOSYS If the functionality is not available.
+ * @retval -errno Other negative errno, result of resuming the device.
  */
-void pm_device_disable(const struct device *dev);
+int pm_device_disable(const struct device *dev);
 
 /**
- * @brief Call device resume synchronously based on usage count
+ * @brief Resume a device based on usage count.
  *
- * Called by a device driver to mark the device as being used. It
- * will bring up or resume the device if it is in suspended state
- * based on the device usage count. This call is blocked until the
- * device PM state is changed to resume.
+ * This function will increase device usage count. If the device is suspended,
+ * it will be resumed. In case of failure, usage count and device state will be
```

Comment:
I would reorder the functions effects here:
- First: "If the device is suspended, it will be resumed." > This is the primary effect expected by the caller
- Then: "This function will increase device usage count." > This is mostly interesting to the pm system

Reply:
Adjusted documentation with a better description, let me know if it's better now